### PR TITLE
Fix issue when keytab contains old and new versions of keys

### PR DESCRIPTION
**Pull request by Zubastic**
Fix for https://github.com/dotnet/Kerberos.NET/issues/165 issue

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/Kerberos.NET?pullRequest=166) <br/>All CLA requirements met.

**Comment by SteveSyfuhs** [comment]
This looks really good. Would it be possible to add a unit test to verify the behavior?

File: Kerberos.NET/Entities/Krb/KrbPrincipalName.cs [hunk 1]
```diff
@@ -155,20 +155,27 @@ private static string MakeFullName(IEnumerable<string> names, PrincipalNameType
...
+            switch (obj)
             {
-                return false;
-            }
-
-            if (!(obj is KrbPrincipalName other))
-            {
-                return false;
-            }
+                case null:
```
Comment (gfoidl) [inline Kerberos.NET/Entities/Krb/KrbPrincipalName.cs]: Can this be handled by `default`?

Comment (Zubastic) [reply Kerberos.NET/Entities/Krb/KrbPrincipalName.cs]: Yeah, fixed. Thanks.

**Comment by SteveSyfuhs** [comment]
And accepted!

**Comment by aptukovainur** [comment]
Why this PR was reversed in https://github.com/dotnet/Kerberos.NET/commit/0b83e6b2e4183e17daece5b3cf12187a929b1771#diff-d01ee0fe161d0959bb2cef60df8352c6f36e348b6518da40e9a9e75350105552?

it doesn't work again :(

**Comment by SteveSyfuhs** [comment]
Fixed. Next nuget package will have this fix. Should be available in ~20 minutes.

**Pull request merged**